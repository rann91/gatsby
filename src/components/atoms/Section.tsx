/* @jsx jsx */
import { PropsWithChildren } from 'react'
import { Box, jsx, SxProps } from 'theme-ui'

type Props = SxProps &
  PropsWithChildren<{
    hasTitle?: boolean
  }>

const Section = ({ hasTitle, children, ...rest }: Props) => (
  <Box as={hasTitle ? 'section' : 'div'} mb={[7, null, null, 8]} {...rest}>
    {children}
  </Box>
)

export default Section
