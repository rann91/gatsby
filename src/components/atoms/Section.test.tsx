import { render } from '@testing-library/react'
import React from 'react'
import Section from './Section'

describe('Section', () => {
  it('renders correctly', () => {
    const { container } = render(<Section />)

    expect(container).toMatchSnapshot()
  })
})
