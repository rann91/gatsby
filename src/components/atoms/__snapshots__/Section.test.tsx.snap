// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `
<div>
  <div
    class="css-1ptauik"
    data-testid="section"
  />
</div>
`;
